### PR TITLE
addremote repo

### PR DESCRIPTION
**Pull request by yifeijin**
# Description
This PR adds a new field `remoteRepo` to the CSM object definition. It's going to download the zip file from the repo and install driver. WIP still need to add certification for the zip file to be downloaded.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Install driver

### CODE REVIEW

File: api/v1alpha1/types.go [hunk 1]
```diff
@@ -129,6 +129,13 @@ type PodStatus struct {
 	Failed    string `json:"failed,omitempty"`
 }
 
+// RemoteRepo to download yamls for driver
+type RemoteRepo struct {
+	Url            string `json:"url,omitempty"`
```
Comment (nb950) [inline api/v1alpha1/types.go:134]: repository 
instead of
url

Comment (nb950) [inline api/v1alpha1/types.go:134]: repository 
instead of
url

**Review by nb950 — APPROVED** [review]

**Review by jooseppi-luna — APPROVED** [review]

File: samples/storage_csm_powerscale.yaml [hunk 2]
```diff
@@ -4,6 +4,10 @@ metadata:
   name: test-isilon
   namespace: test-isilon
 spec:
+  remoteRepo:
+    repository: "https://amaas-eos-mw1.cec.lab.emc.com:5036/artifactory/csi-driver-helm-virtual"
```
Comment (rensyct) [inline samples/storage_csm_powerscale.yaml:8]: @yifeijin, we should not be having references to any of the internal repo
FYI @bharathsreekanth

**Comment by bharathsreekanth** [comment]
@yifeijin - Please don't merge PRs if there are open comments that have NOT been addressed.

**Pull request merged**